src/pkg_config_gen.rs

```diff
@@ -307,6 +306,102 @@ mod test {
         );
         pkg.add_lib("-lbar").add_cflag("-DFOO");
 
-        println!("{:?}\n{}", pkg, pkg.render());
+        let expected = concat!(
+            "prefix=/usr/local\n",
+            "exec_prefix=${prefix}\n",
+            "libdir=${exec_prefix}/lib\n",
+            "includedir=${prefix}/include\n",
+            "\n",
+            "Name: foo\n",
+            "Description: \n",
+            "Version: 0.1\n",
+            "Libs: -L${libdir} -lfoo -lbar\n",
+            "Cflags: -I${includedir} -DFOO\n",
+            "Requires: somelib, someotherlib\n",
+            "Requires.private: someprivatelib >= 1.0\n",
+        );
+
+        assert_eq!(expected, pkg.render());
+    }
+
+    mod test_canonicalize {
+        use super::canonicalize;
```

Comment:
It needs tests for "\a\b\c" and such paths.

Reply:
But only on windows...

Reply:
on windows, `a\b\c` gets parsed as a single normal component. Also pkgconfig is just for unix systems right, so windows paths don' t make a whole lot of sense even when cross-compiling. Maybe WSL causes complications?

Reply:
pkgconfig exists and is used also on windows :) See https://github.com/lu-zero/cargo-c/issues/210 on why